code/game/objects/items/devices/radio/radio.dm

```diff
@@ -159,6 +159,10 @@
 	for(var/channel_name in channels)
 		secure_radio_connections[channel_name] = add_radio(src, GLOB.radiochannels[channel_name])
 
+	if(!listening)
+		set_listening(TRUE)
+		set_listening(FALSE)
```

Comment:
This just screams at me as a bad fix, you should put the specific caching part of the function instead of just turning it on and off again (remove_radio_all(src) if I understood your description of the issue correctly)

Reply:
https://github.com/user-attachments/assets/e6f586ad-44c3-44db-9c64-6e0b0e76c9d5

this is a video of me doing it with encryption keys, granted it is downstream, but this is an issue here too

Reply:
Your suggestion DID work though I appreciate it